Dart/src/com/jetbrains/lang/dart/ide/runner/server/OpenDartObservatoryUrlAction.java

```diff
@@ -34,20 +36,25 @@ public void setUrl(@NotNull final String url) {
     myUrl = url;
   }
 
+  public void setWsUrl(@NotNull final String wsUrl) {
+    myWebsocketUrl = wsUrl;
+  }
+
   @Override
   public @NotNull ActionUpdateThread getActionUpdateThread() {
     return ActionUpdateThread.BGT;
   }
 
   @Override
   public void update(@NotNull final AnActionEvent e) {
-    e.getPresentation().setEnabled(myUrl != null && myIsApplicable.compute());
+    e.getPresentation().setEnabled(myWebsocketUrl != null && myIsApplicable.compute());
   }
 
   @Override
   public void actionPerformed(@NotNull final AnActionEvent e) {
-    if (myUrl != null) {
-      openUrlInChromeFamilyBrowser(myUrl);
+    if (myWebsocketUrl != null) {
+      final String url = e.getProject().getService(DevToolsService.class).getDevToolsInstance().getUrl() + "/?uri=" + myWebsocketUrl;
```

Comment:
the implementation of the `getDevToolsConsumer` is
```
devToolsFuture.get()
```
This means that if DevTools process has not been started for some reason (we discussed these reasons by email), this call freezes the IDE. 

My thoughts are:
- not to use Future in DevToolsService (BTW, I renamed this class to DartDevToolsService, I hope it's fine)
- getDevToolsInstance() to be nullable and return the current result instantly
- keep compatibility with older SDKs, for which we don't start DTD+DT, so I mean keep using old behavior of the OpenDartObservatoryUrlAction when the new behaviour is not applicable

BTW, maybe renale DevToolsInstance to DartDevToolsInfo? It's just a data class for host and port.